maintainer_reports/graphql_requests.py

```diff
@@ -0,0 +1,255 @@
+    of the JSON documents in the file.  Alterations made to the table to add 
+    additonal columns will be lost.
+    """
+
+    client = get_bq_client()
+
+    dataset_id = " pull-request-reporting.open_edx_github "
+    table_id = "pull-request-reporting.open_edx_github.open_pull_requests"
+
+    job_config = bigquery.LoadJobConfig()
+    job_config.source_format = bigquery.SourceFormat.NEWLINE_DELIMITED_JSON
+    job_config.autodetect = True
+    job_config.write_disposition = "WRITE_TRUNCATE"
+
+    with open(filename, "rb") as source_file:
+        job = client.load_table_from_file(
+            source_file,
+            table_id,
+            location="US",
+            job_config=job_config,
+        )
+
+    job.result()
+
+    click.echo(f"Loaded {job.output_rows} rows into {dataset_id}:{table_id}.")
+
+
+def create_temp_table(jsonl_file, table_name):
+    """
+    Creates and populuates a temp table from that JSON Lines formated file.
+    """
+
+    client = get_bq_client()
+
+    dataset_id = "pull-request-reporting.open_edx_github"
+    table_id = "pull-request-reporting.open_edx_github." + table_name
+
+    job_config = bigquery.LoadJobConfig()
+    job_config.source_format = bigquery.SourceFormat.NEWLINE_DELIMITED_JSON
+    job_config.autodetect = True
+    job_config.create_disposition = "CREATE_IF_NEEDED"
+    job_config.write_disposition = "WRITE_TRUNCATE"
+
+    with open(jsonl_file, "rb") as source_file:
+        job = client.load_table_from_file(
+            source_file,
+            table_id,
+            location="US",
+            job_config=job_config,
+        )
+
+    job.result()
+
+    click.echo(f"Loaded {job.output_rows} rows into {dataset_id}:{table_id}.")
+
+
+def upsert_closed_pulls(table_name):
+    """
+    Merges records from a temporary table into a destination. Currently
+    uses the permalink as a pseudo key.
+    """
+
+    query = UPSERT_CLOSED_PULLS.replace("_TEMP_TABLE_NAME_", table_name)
+    query_job = execute_dml_bigquery(query)
+    click.echo(f"Upserting closed pulls affected {query_job.num_dml_affected_rows} rows.")
+
+
+def update_cols_bq(target_table_id):
+    """
+    When the table is recreated via the write disposition WRITE_TRUNCATE
+    the schema is discerned from the JSONL document.  Two columns to track
+    maintainers are added after the fact.
+    """
+    client = get_bq_client()
+
+    table = client.get_table(target_table_id)
+
+    original_schema = table.schema
+    new_schema = original_schema[:]  
+    new_schema.append(bigquery.SchemaField("maintainer", "STRING"))
+    new_schema.append(bigquery.SchemaField("maintainer_organization", "STRING"))
+
+    table.schema = new_schema
+    table = client.update_table(table, ["schema"])
+
+
+def update_maintainers_bq(target_table):
+    """
+    Populated the maintainer related columns by joining against
+    an external table which is a Google Sheet.
```

Comment:
Should we link to this sheet in the Readme?

Reply:
Done.